test/e2e/performanceprofile/functests/1_performance/topology_manager.go

```diff
@@ -45,4 +49,88 @@ var _ = Describe("[rfe_id:27350][performance]Topology Manager", Ordered, func()
+			Expect(workerRTNodes).ToNot(BeEmpty(), "No RT worker node found!")
+			targetNode = &workerRTNodes[0]
+
+			initialProfile = profile.DeepCopy()
+
+			By("Adding the DRA resource management annotation to the profile")
+			if profile.Annotations == nil {
+				profile.Annotations = make(map[string]string)
+			}
+			profile.Annotations[performancev2.PerformanceProfileDRAResourceManagementAnnotation] = "true"
+
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			poolName := poolname.GetByProfile(context.TODO(), profile)
+			By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+			profilesupdate.WaitForTuningUpdating(context.TODO(), profile)
+
+			By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+			profilesupdate.WaitForTuningUpdated(context.TODO(), profile)
+
+			DeferCleanup(func() {
+				By("Reverting the performance profile")
+				profiles.UpdateWithRetry(initialProfile)
+
+				poolName := poolname.GetByProfile(context.TODO(), initialProfile)
+				By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+				profilesupdate.WaitForTuningUpdating(context.TODO(), initialProfile)
+
+				By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+				profilesupdate.WaitForTuningUpdated(context.TODO(), initialProfile)
+
+				By("Verifying managers are restored to their original configuration")
+				kubeletConfig, err := nodes.GetKubeletConfig(context.TODO(), targetNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Verifying CPU manager policy is restored to static")
+				Expect(kubeletConfig.CPUManagerPolicy).To(Equal("static"), "CPUManagerPolicy should be 'static' after revert, got %q", kubeletConfig.CPUManagerPolicy)
+
+				By("Verifying Topology manager policy is restored")
+				expectedTopologyPolicy := kubeletconfigv1beta1.BestEffortTopologyManagerPolicy
+				if initialProfile.Spec.NUMA != nil && initialProfile.Spec.NUMA.TopologyPolicy != nil {
+					expectedTopologyPolicy = *initialProfile.Spec.NUMA.TopologyPolicy
+				}
+				Expect(kubeletConfig.TopologyManagerPolicy).To(Equal(expectedTopologyPolicy), "TopologyManagerPolicy should be %q after revert, got %q", expectedTopologyPolicy, kubeletConfig.TopologyManagerPolicy)
+
+				By("Verifying Memory manager policy is restored")
+				// Memory manager is set to Static only for restricted or single-numa-node topology policies
+				if expectedTopologyPolicy == kubeletconfigv1beta1.RestrictedTopologyManagerPolicy ||
+					expectedTopologyPolicy == kubeletconfigv1beta1.SingleNumaNodeTopologyManagerPolicy {
+					Expect(kubeletConfig.MemoryManagerPolicy).To(Equal("Static"), "MemoryManagerPolicy should be 'Static' after revert, got %q", kubeletConfig.MemoryManagerPolicy)
+				}
+			})
+		})
+
+		It("should disable CPU, Memory, and Topology managers when DRA annotation is set", func() {
```

Comment:
Would also be nice to validate that the original manager settings are restored accurately when the annotation is removed.

Reply:
Thanks, I agree, we do https://github.com/Tal-or/cluster-node-tuning-operator/blob/nri_plugin_deployment/test/e2e/performanceprofile/functests/1_performance/topology_manager.go#L106-#L117